containers/daap-landing-to-raw/tests/unit/validation_test.py

```diff
@@ -0,0 +1,169 @@
+import pytest
+from validation import DataInvalid, type_is_compatible, validate_data_against_schema
+
+
+@pytest.mark.parametrize(
+    "registered_type,inferred_type,expected",
+    [
+        ("tinyint", "tinyint", True),
+        ("smallint", "smallint", True),
+        ("integer", "integer", True),
+        ("int", "int", True),
+        ("bigint", "bigint", True),
+        ("float", "float", True),
+        ("double", "double", True),
+        ("decimal", "decimal", True),
+        ("string", "string", True),
+        ("boolean", "boolean", True),
+        ("char", "char", True),
+        ("timestamp", "timestamp", True),
+        ("date", "date", True),
+        ("array", "array", True),
+        ("struct", "struct", True),
+        ("garbage", "int", False),
+        ("int", "garbage", False),
+        ("tinyint", "smallint", True),
+        ("tinyint", "int", True),
+        ("tinyint", "bigint", True),
+        ("tinyint", "float", False),
+        ("tinyint", "double", False),
+        ("tinyint", "decimal", False),
+        ("tinyint", "string", False),
+        ("tinyint", "boolean", False),
+        ("tinyint", "timestamp", False),
+        ("tinyint", "date", False),
+        ("smallint", "tinyint", True),
+        ("smallint", "int", True),
+        ("smallint", "bigint", True),
+        ("smallint", "float", False),
+        ("smallint", "double", False),
+        ("smallint", "decimal", False),
+        ("smallint", "string", False),
+        ("smallint", "boolean", False),
+        ("smallint", "timestamp", False),
+        ("smallint", "date", False),
+        ("int", "tinyint", True),
+        ("int", "smallint", True),
+        ("int", "bigint", True),
+        ("int", "float", False),
+        ("int", "double", False),
+        ("int", "decimal", False),
+        ("int", "string", False),
+        ("int", "boolean", False),
+        ("int", "timestamp", False),
+        ("int", "date", False),
+        ("bigint", "tinyint", True),
+        ("bigint", "smallint", True),
+        ("bigint", "int", True),
+        ("bigint", "float", False),
+        ("bigint", "double", False),
+        ("bigint", "decimal", False),
+        ("bigint", "string", False),
+        ("bigint", "boolean", False),
+        ("bigint", "timestamp", False),
+        ("bigint", "date", False),
+        ("float", "tinyint", True),
+        ("float", "smallint", True),
+        ("float", "int", True),
+        ("float", "bigint", True),
+        ("float", "double", True),
+        ("float", "decimal", True),
+        ("float", "string", False),
+        ("float", "boolean", False),
+        ("float", "timestamp", False),
+        ("float", "date", False),
+        ("double", "tinyint", True),
+        ("double", "smallint", True),
+        ("double", "int", True),
+        ("double", "bigint", True),
+        ("double", "double", True),
+        ("double", "decimal", True),
```

Comment:
Just FYI these tests are quite slow because there's 100 of them. We could always remove some cases if we want to speed this up